attractions/file-input/file-dropzone.svelte

```diff
@@ -26,20 +27,27 @@
     wrongType = [...e.dataTransfer.items].some(file => !accepted(accept, file));
   }
 
-  function acceptUpload(e) {
+  async function acceptUpload(e) {
     for (let file of (e.dataTransfer || e.target).files) {
-      if (accepted(accept, file)) {
-        files.push(file);
-      }
+      try {
+        if (typeof beforeChange === 'function') {
+          await beforeChange(file);
+        }
+
+        if (accepted(accept, file)) {
+          files.push(file);
+        }
+      } catch (e) {}  // eslint-disable-line no-empty
```

Comment:
Wouldn't it be a little neater if this try-catch was moved to surround just the `beforeChange` call, with the `catch` block having just `continue`?